### ISSUE TITLE
Auto include files in index.html.

### ISSUE THREAD

**Issue by AnkurIngale** [processing/p5.js-web-editor#948]
#### Nature of issue?

- New feature request

#### New feature details:
I feel on adding a new file, it should automatically be included in index.html.
Views?

**Comment by siddhant1** [comment]
Don't think it's that important imho

**Comment by AnkurIngale** [comment]
Yeah but it's not a bug, it's a feature which can be implemented anytime 😄

**Comment by GaurangTandon** [comment]
I don't think that should happen. Not all files always need to be included in index.html, for example, css files that are imported by other css files.

**Comment by catarak** [comment]
i like this idea and i've definitely talked about this before! i want to think it through though. some thoughts/questions:
- how much "magic" do we want the web editor to have? i'm hesitant to introduce things that folks have to unlearn, i.e. when adding a file in another text editor, they would have to add it manually. but maybe this is okay. 
- if a user then removes the file from the `index.html`, would it get put back in?

other pros/cons?

**Comment by AnkurIngale** [comment]
I feel we can add them the first time the user makes a file and give the user a notification that the newly created file has been added. If the user wishes to remove it, he can remove it later.

**Comment by catarak** [comment]
i like the idea of it being an explicit thing that's happening. not quite sure what the UI/UX for this would look like.

**Comment by bmoren** [comment]
I feel like it's a very processing like behavior to have it auto-add to the index.html, and I like that. But as this is the web (and linking tons of files is so common) I feel that the user should at least be aware that something magic happened. 

If tabs are going to be a thing at some point, maybe there is something there that could happen there? maybe the new file opens and also the index.html opens in a tab and there is a comment in the HTML above the added file which says  

I think a further discussion is what files should be auto added, I'd suggest that only making additional .js files (as they are common for breaking classes out) might be the way to go. IMO, All others (CSS files, etc.) should just be ignored and let the user add/link them since that's a particularly out of the 'typical' scope of common usage of p5 and a more advanced topic where the user should probably know or learn about linking at that point anyway.

**Comment by catarak** [comment]
that's a good point about it being processing-like behavior. it's hard to know with this project what is okay magic and what is too magical, which i think is related to the difference between JS and Java. 

i think only JS and CSS files could even be auto-added anyway (because... can you add other types of files to HTML?), which is a nice constraint.

**Comment by bmoren** [comment]
I think you're right, css files are the only other thing to link. But, I'd be hesitant to even link those since it's not that uncommon to link dozens of css files for fonts, modules, and other styles & this is not really intended to be a 'web' development ide, but a p5.js editor, where typically the CSS is very light, if not non-existent all together!

**Comment by AnkurIngale** [comment]
@bmoren Even I feel only js files should be added. CSS files are generally not needed and if they are required, the user can add it himself. But for the js files, I feel we should automatically add it. Giving user a notif will do it.

**Comment by catarak** [comment]
i'm definitely warming up to this issue! maybe simply adding new JS files to the `index.html`, and adding a notification is enough.

**Comment by ashu8912** [comment]
Hey @catarak whats your take on this issue now should we add this feature?

**Comment by I-Omnibus** [comment]
_[Here](https://editor.p5js.org/Omnibus/sketches/ajcBWKWbT)_'s a possible workaround to load dependencies without editing index.html

```Javascript
function preload() {
  Dependencies.resolve(["load_me"], /*bypassCache*/ true);
}
```
